ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -80,7 +81,9 @@ public JoinNode(@JsonProperty("id") final PlanNodeId id,
                   @JsonProperty("rightType") final DataSource.DataSourceType rightType) {
 
     // TODO: Type should be derived.
-    super(id);
+    super(id, (leftType == DataSourceType.KTABLE && rightType == DataSourceType.KTABLE)
```

Comment:
Can't this just be simplified to leftType ?

Reply:
Yes, but this is more stricter check!